doc/mkdocs/docs/features/macros.md

```diff
@@ -79,6 +79,10 @@ The first parameter is the name of the class/struct, and all remaining parameter
 
 See [Simplify your life with macros](arbitrary_types.md#simplify-your-life-with-macros) for an example.
 
+## `NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT(type, member...)`
+
+This macro is similar to `NLOHMANN_DEFINE_TYPE_INTRUSIVE` but will throw no exceptions. When converting JSON object to type object, if any field is missing in JSON object, it will use default value constructed by the type itself.
```

Comment:
I'm thinking this should call out that it requires that the type be default constructible and that it constructs an object inside the `from_json()` function.  I think it may also still throw exceptions if the types don't match.

Perhaps something like this:

This macro is similar to `NLOHMANN_DEFINE_TYPE_INTRUSIVE`.  It will not throw an exception in `from_json()` due to a missing value in the JSON, but can throw due to a mismatched type.  In order to support that it requires that the type be default constructible.  The `from_json()` function default constructs an object and uses its values as the defaults when calling the `value()` function.

Reply:
Sounds better, thanks. Already updated.